### PR TITLE
build: Cocoapods integration

### PR DESCRIPTION
**Pull request by tomaszczart**
This PR introduces CocoaPods for KMM project.

_Unfortunately we cannot upgrade to Kotlin 1.4.20, because Jetpack Compose is not compatible with it yet._

### CODE REVIEW

File: domain/build.gradle.kts [hunk 1]
```diff
@@ -21,18 +22,20 @@ kotlin {
...
-            framework {
-                baseName = "domain"
-            }
-        }
+    ios()
+
+    cocoapods {
+        summary = "SongApp domian library."
+        homepage = "https://github.com/tooploox/KMMSongApp"
+        ios.deploymentTarget = "13.2"
```
Comment (wzieba) [inline domain/build.gradle.kts:30]: 14.2 is the newest I think
```suggestion
        ios.deploymentTarget = "14.2"
```

File: .gitignore [hunk 2]
```diff
@@ -4,4 +4,10 @@ build/
 .idea/
 .DS_STORE
 local.properties
+*.hprof
 
+## Xcode User settings
+xcuserdata/
+
+## Pods
+Pods/
```
Comment (tomaszczart) [inline .gitignore:13]: I think that there is no crucial benefits of keeping `Pods/` even though Cocoapods recommends to keep Pods directory under source control- [more here](https://guides.cocoapods.org/using/using-cocoapods.html#should-i-check-the-pods-directory-into-source-control). Idk if it is a good comparison, but it is a bit like keeping Gradle cache under source control. WDYT?

File: domain/build.gradle.kts [hunk 3]
```diff
@@ -21,18 +22,20 @@ kotlin {
...
-                baseName = "domain"
-            }
-        }
+    ios()
+
+    cocoapods {
+        summary = "SongApp domian library."
+        homepage = "https://github.com/tooploox/KMMSongApp"
+        ios.deploymentTarget = "14.2.1"
+        podfile = project.file("../iosApp/Podfile")
```
Comment (tomaszczart) [inline domain/build.gradle.kts:31]: It synchronize our Xcode project with Kotlin Pod dependencies by calling `pod install` for our `Podfile`.

File: iosApp/iosApp.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist [hunk 4]
```diff
@@ -2,13 +2,7 @@
...
-		<key>iosApp.xcscheme_^#shared#^_</key>
-		<dict>
-			<key>orderHint</key>
-			<integer>0</integer>
-		</dict>
-	</dict>
+	<key>IDEDidComputeMac32BitWarning</key>
+	<true/>
 </dict>
 </plist>
```
Comment (tomaszczart) [inline iosApp/iosApp.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist:8]: > Xcode 9.3 adds a new IDEWorkspaceChecks.plist file to a workspace's shared data, to store the state of necessary workspace checks. Committing this file to source control will prevent unnecessary rerunning of those checks for each user opening the workspace.

**Comment by wzieba** [comment]
Can we ask someone from @tooploox/ios for help with reviewing this PR? 🙏

File: iosApp/Podfile [hunk 5]
```diff
@@ -0,0 +1,7 @@
+use_frameworks!
+
+platform :ios, '14.2.1'
```
Comment (mateusznadolski) [inline iosApp/Podfile:3]: You probably want to support older version of iOS too

File: .gitignore (hunk 2 above)
Comment (tomaszczart) [reply .gitignore:13]: @mateusznadolski what do you usually put into `.gitignore`?

Comment (mateusznadolski) [reply .gitignore:13]: We usually include `Pods` in the repo. But both options are fine

**Pull request closed without merging**